packages/shared-internal/src/clients/novu-base-client.ts

```diff
@@ -0,0 +1,170 @@
+// Base HttpError class with response field
+export class HttpError extends Error {
```

Comment:
Why do we need this? Since we are using NestJS we need to stick to its exception system as much as possible. This feels like duplication

Reply:
it belong to the client, not nest, you don't thorw nest errors to the client.